### PR TITLE
docs: API for local user management

### PR DESCRIPTION
**Pull request by kjaskiewiczz**
Description can be found here: https://tracker.mender.io/browse/MEN-828
@maciejmrowiec @mendersoftware/rndity

### CODE REVIEW

File: docs/management_api.yml [hunk 1]
```diff
@@ -58,8 +58,182 @@ paths:
...
+            $ref: "#/definitions/UserNew"
+        - name: Authorization
+          in: header
+          required: true
+          type: string
+          format: Bearer [token]
+          description: Contains the JWT token issued by the User Administration and Authentication Service.
+      responses:
+        204:
+          description: User created.
```
Comment (bboozzoo) [inline docs/management_api.yml]: Should we return a `Location` with URL to created user profile?

File: docs/management_api.yml [hunk 2]
```diff
@@ -69,10 +243,50 @@ definitions:
...
+    properties:
+      email:
+        description: A unique email address.
+        type: string
+      id:
+        description: User Id.
+        type: string
+      updated_ts:
+        description: |
+            Server-side timestapm of the last user information update.
```
Comment (bboozzoo) [inline docs/management_api.yml]: `s/timestapm/timestamp/`

File: docs/management_api.yml [hunk 3]
```diff
@@ -69,10 +243,50 @@ definitions:
...
+        type: string
+        format: datetime
```
Comment (bboozzoo) [inline docs/management_api.yml]: It's `date-time` http://swagger.io/specification/#data-types-12

File: docs/management_api.yml [hunk 4]
```diff
@@ -81,6 +295,10 @@ definitions:
       error:
         description: Description of the error.
         type: string
+      request_id:
```
Comment (bboozzoo) [inline docs/management_api.yml]: IIRC this comes back to the use as `X-MEN-RequestID` header

**Comment by maciejmrowiec** [comment]
some of those operations look like need to be restricted to administrative roles, should the description mention this already? or something for later ?

File: docs/management_api.yml (hunk 1 above)
Comment (kjaskiewiczz) [inline docs/management_api.yml]: done

File: docs/management_api.yml (hunk 2 above)
Comment (kjaskiewiczz) [inline docs/management_api.yml]: done

File: docs/management_api.yml (hunk 3 above)
Comment (kjaskiewiczz) [inline docs/management_api.yml]: done

File: docs/management_api.yml (hunk 4 above)
Comment (kjaskiewiczz) [inline docs/management_api.yml]: done

**Comment by kjaskiewiczz** [comment]
There is no roles in the system yet, so it's something for later, imo.

**Comment by mchalski** [comment]
@maciejmrowiec I think it's for later, everyone's a superuser for now and we have no mention of roles anywhere yet.

File: docs/management_api.yml [hunk 5]
```diff
@@ -58,8 +59,186 @@ paths:
...
+          description: User id.
+          required: true
+        - name: Authorization
+          in: header
+          required: true
+          type: string
+          format: Bearer [token]
+          description: Contains the JWT token issued by the User Administration and Authentication Service.
+      responses:
+        204:
```
Comment (mchalski) [inline docs/management_api.yml]: 204?

**Comment by mchalski** [comment]
@kjaskiewiczz have you seen https://tracker.mender.io/browse/MEN-828?focusedCommentId=80121&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-80121? (add a `created_ts` field, should be simple enough`

File: docs/management_api.yml (hunk 5 above)
Comment (kjaskiewiczz) [inline docs/management_api.yml]: yup

**Comment by kjaskiewiczz** [comment]
@mchalski as you wish (or as @michaelatmender wish)

File: docs/management_api.yml (hunk 4 above)
Comment (kjaskiewiczz) [inline docs/management_api.yml]: The RestErrWithLogMsg from go-lib-micro/rest_utils is adding request_id to response body.

Comment (bboozzoo) [inline docs/management_api.yml]: I suppose that's ok too.

**Comment by bboozzoo** [comment]
merge?

**Pull request merged**